python/src/sensor.py

```diff
@@ -0,0 +1,139 @@
+import time
+from datetime import datetime
+
+import uuid
+import random
+import dognames
+import names
+import argparse
+from db import config
+from db.client import ScyllaClient
+
+
+def generate_random_pet_name():
+    return dognames.male() if random.randint(1, 2)==1 else dognames.female()
+
+
+def generate_random_measurement_value(sensor_type):
+    if sensor_type == "T":
+        return 101 + random.uniform(0, 10) - 4;
+    elif sensor_type == "L":
+      return 35 + random.uniform(0, 5) - 2
+    return random.uniform(0, 20)
+
+
+def generate_random_measurement(sensor):
+    return {
+        "sensor_id": sensor["sensor_id"],
+        "ts": datetime.now(),
+        "value": generate_random_measurement_value(sensor["type"])
+    }
+  
+  
+def create_owner():
+    return {
+        "owner_id": uuid.uuid4(),
+        "address": "home",
+        "name": names.get_full_name() # generate random person's name
+    }
+
+
+def create_pet(owner):
+    return {
+        "owner_id": owner["owner_id"],
+        "pet_id": uuid.uuid4(),
+        "address": "home",
+        "age": random.randint(1, 20),
+        "name": generate_random_pet_name(),
+        "weight": random.uniform(5.0, 15.0)
+    }
+
+
+def create_sensors(pet):
+    return [
+        {
+            "pet_id": pet["pet_id"],
+            "sensor_id": uuid.uuid4(),
+            "type": "T"
+        },
+        {
+            "pet_id": pet["pet_id"],
+            "sensor_id": uuid.uuid4(),
+            "type": "L"
+        }
+    ]
+
+def insert_pet_static_data(client: ScyllaClient, pet, owner, sensors):
+    client.insert_data(table="carepet.pet", data_dict=pet)
+    client.insert_data(table="carepet.owner", data_dict=owner)
+    for sensor in sensors:
+        client.insert_data(table="carepet.sensor", data_dict=sensor)
+    
+    
+def generate_measurements(sensors, count=2, delay=1):
+    measurements_list = []
+    for i in range(0, int(count)):
+        for sensor in sensors:
+            measurement = generate_random_measurement(sensor)
+            measurements_list.append(measurement)
+            
+            print(f"sensor # {sensor['sensor_id']} type {sensor['type']},",
+                  f"new measure: {measurement['value']},",
+                  f"ts: {measurement['ts']}")
+        time.sleep(delay)
```

Comment:
you can pick random time in the future/past, it's a bit weird to sleep just to generate timestamps.

Reply:
Good idea. But my understanding is that this script is supposed to simulate a real-world data ingestion script. If we use random timestamps then real-time queries (those that include eg. `now()` or similar) wouldn't work.

Reply:
As you said it's a simulation of timed data, you can put what you want as long as it's a correct date.

It would enable you to apply more randomess as needed, with affecting the time it would take to execute this code.